java/ql/src/semmle/code/java/frameworks/Flexjson.qll

```diff
@@ -0,0 +1,42 @@
+/**
+ * Provides classes for working with the Flexjson framework.
+ */
+
+import java
+
+/** The class `flexjson.JSONDeserializer`. */
+class FlexjsonDeserializer extends RefType {
+  FlexjsonDeserializer() { this.hasQualifiedName("flexjson", "JSONDeserializer") }
+}
+
+/** The class `flexjson.JSONSerializer`. */
+class FlexjsonSerializer extends RefType {
+  FlexjsonSerializer() { this.hasQualifiedName("flexjson", "JSONSerializer") }
+}
+
+/** The deserialization method `deserialize`. */
+class FlexjsonDeserializeMethod extends Method {
+  FlexjsonDeserializeMethod() {
+    this.getDeclaringType().getSourceDeclaration().getASourceSupertype*() instanceof
+      FlexjsonDeserializer and
+    this.getName().matches("deserialize%")
```

Comment:
Is this just `deserialize` and `deserializeInto`? If so use an explicit list instead of a match.

Reply:
Changed